lib/shared/drive-constraints.js

```diff
@@ -208,3 +208,85 @@ exports.isDriveValid = (drive, image) => {
 exports.isDriveSizeRecommended = (drive, image) => {
   return _.get(drive, [ 'size' ], UNKNOWN_SIZE) >= _.get(image, [ 'recommendedDriveSize' ], UNKNOWN_SIZE);
 };
+
+const feedbackMessages = {
+  NOTREC: 'Not Recommended',
+  TOOSMALL: 'Too Small For Image',
+  LOCKED: 'Locked',
+  SYSTEM: 'System Drive',
+  SOURCE: 'Drive Contains Image'
+};
+const FEEDBACK_ERROR = 'error';
+const FEEDBACK_WARNING = 'warning';
+
+/**
+ * @summary Get drive feedback in an object, if any
+ * @function
+ * @public
+ *
+ * @description
+ * If the drive-image combination results in a warning or error, this function
+ * returns an object containing the feedback type ('error', 'warning') and the
+ * feedback message.
+ *
+ * @param {Object} drive - drive
+ * @param {Object} image - image
+ * @returns {Object} feedback object
+ *
+ * @example
+ * const drive = {
+ *   device: '/dev/disk2',
+ *   name: 'My Drive',
+ *   size: 4000000000
+ * };
+ *
+ * const image = {
+ *   path: 'rpi.img',
+ *   size: 2000000000
+ *   recommendedDriveSize: 4000000000
+ * });
+ *
+ * const { type, message } = DriveConstraints.getDriveFeedback(drive, image);
+ *
+ * if (type === 'warning') {
+ *   // still allow flashing
+ * } else if (type === 'error') {
+ *   // disallow flashing
+ * }
+ */
+exports.getDriveFeedback = (drive, image) => {
```

Comment:
That's where I've ended up now. The test currently returns "Too Small For Image", and it's not completely wrong considering the drive is nonexistent. `OK` sounds weird here since it's really not okay, but making another message like `EMPTY` solely for this might not be ideal either? Maybe we can change from `OK` to sound less validating and more neutral?